### PR TITLE
fix: [SRM-13807]: Making call to update analysis type null safe

### PR DESCRIPTION
**Pull request by dhruv-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43317)

### CODE REVIEW

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/activity/services/impl/ActivityServiceImpl.java [hunk 1]
```diff
@@ -156,6 +156,7 @@ public Optional<Activity> getAnyDemoDeploymentEvent(MonitoredServiceParams monit
 
   @Override
   public String upsert(Activity activity) {
+    log.info("Creating  Activity {}", activity.toString());
```
Comment (n00bitax) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/activity/services/impl/ActivityServiceImpl.java:159]: tostring is not needed theres issue of null safety too, we should add object only.
Also I think log with DB state changes will help , add it for create too.

**Review by n00bitax — DISMISSED** [review]
> Reviewed 3 of 6 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43317)* status: :shipit: complete! ✓

**Comment by AbhijithHarness** [comment]
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/activity/services/impl/ActivityServiceImpl.java` line 159 at r2](https://reviewable.io/reviews/harness/harness-core/43317#-NN8Aj4DpQSny_tmIwe5-r2-159:-NNAtKvK5Oag1E0K8q5M:bx5gf60) ([raw file](https://github.com/harness/harness-core/blob/0321682a34f855b24f8bdac46bf4e6c3c67aa9f5/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/activity/services/impl/ActivityServiceImpl.java#L159)):*
<details><summary><i>Previously, n00bitax (Ansuman Satapathy) wrote…</i></summary><blockquote>

tostring is not needed theres issue of null safety too, we should add object only.
Also I think log with DB state changes will help , add it for create too.
</blockquote></details>

Lets not print the entire activity. There are yaml diffs also present for some activities that the activity can be too large. Just print getEntityKeyLongString() value of it.

**Review by AbhijithHarness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43317)* status: 1 change requests. 1. unresolved discussions. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CVNGStep.java` line 128 at r2](https://reviewable.io/reviews/harness/harness-core/43317#-NNAu_x48hrVPpleh46A:-NNAu_x50U6ljKcEAAw9:bx54bf7) ([raw file](https://github.com/harness/harness-core/blob/0321682a34f855b24f8bdac46bf4e6c3c67aa9f5/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CVNGStep.java#L128)):*
> ```Java
>     String monitoredServiceTemplateVersionLabel = resolvedCVConfigInfo.getMonitoredServiceTemplateVersionLabel();
>     List<CVConfig> cvConfigs = resolvedCVConfigInfo.getCvConfigs();
>     log.info("ResolvedCVConfigInfo {}", resolvedCVConfigInfo);
> ```

Just print CV Config Id. Dont see much use of these, as it will be anyway saved in the verificationJobInstance log

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/activity/services/impl/ActivityServiceImpl.java (hunk 1 above)
Comment (dhruv-harness) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/activity/services/impl/ActivityServiceImpl.java]: Addressed

**Review by dhruv-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43317)* status: 5 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CVNGStep.java` line 128 at r2](https://reviewable.io/reviews/harness/harness-core/43317#-NNAu_x48hrVPpleh46A:-NNBBIjWA7n9271BdyPZ:b32w0x) ([raw file](https://github.com/harness/harness-core/blob/0321682a34f855b24f8bdac46bf4e6c3c67aa9f5/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CVNGStep.java#L128)):*
<details><summary><i>Previously, AbhijithHarness (Abhijith Pradeep) wrote…</i></summary><blockquote>

Just print CV Config Id. Dont see much use of these, as it will be anyway saved in the verificationJobInstance log
</blockquote></details>

OK
___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/activity/services/impl/ActivityServiceImpl.java` line 159 at r2](https://reviewable.io/reviews/harness/harness-core/43317#-NN8Aj4DpQSny_tmIwe5-r2-159:-NNBBBDk-SwV8uB4IHTQ:bq76sqt) ([raw file](https://github.com/harness/harness-core/blob/0321682a34f855b24f8bdac46bf4e6c3c67aa9f5/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/activity/services/impl/ActivityServiceImpl.java#L159)):*
<details><summary><i>Previously, dhruv-harness (Dhruv) wrote…</i></summary><blockquote>

Addressed
</blockquote></details>

Removed the log for activity class as there is a log already present at the end

**Review by deepak-harness — APPROVED** [review]
> Reviewed 1 of 6 files at r1, 1 of 3 files at r2, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43317)* status: 1 change requests. 2. unresolved discussions.

**Review by n00bitax — APPROVED** [review]

**Review by AbhijithHarness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43317)* status: 2. unresolved discussions.

**Review by AbhijithHarness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43317)* status: 1. unresolved discussions.

**Review by AbhijithHarness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43317)* status: 1. unresolved discussions.

**Review by n00bitax — APPROVED** [review]
> Reviewed 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43317)* status: 1. unresolved discussions.

**Review by n00bitax — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43317)* status: :shipit: complete! ✓

**Pull request merged**